app/src/main/java/org/simple/clinic/instantsearch/InstantSearchEffectHandler.kt

```diff
@@ -73,6 +77,29 @@ class InstantSearchEffectHandler @AssistedInject constructor(
     }
   }
 
+  private fun checkIfPatientAlreadyHasAnExistingNHID(scheduler: Scheduler): ObservableTransformer<CheckIfPatientAlreadyHasAnExistingNHID, InstantSearchEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .observeOn(schedulers.computation())
+          .flatMap { patientRepository.patientProfile(it.patientId).subscribeOn(scheduler) }
+          .map { patientProfile ->
+            val patientUUID = patientProfile.get().patientUuid
+
+            if (hasNationalHealthID(patientProfile.get()))
+              PatientAlreadyHasAnExistingNHID(patientUUID)
+            else
+              PatientDoesNotHaveAnExistingNHID(patientUUID)
+          }
+    }
+  }
+
+  private fun hasNationalHealthID(patientProfile: PatientProfile): Boolean {
+    return patientProfile
+        .withoutDeletedBusinessIds()
+        .businessIds
+        .any { it.identifier.type == IndiaNationalHealthId }
```

Comment:
We could move this check to a computed property within the `PatientProfile` type.